### DISCUSSION TITLE
browser.switchToFrame

### DISCUSSION THREAD

**Discussion by daganilsenwork** [webdriverio/webdriverio#7322] [Technical Discussion]
Hi. How do I switch to a iframe?
I can see a lot of examples online but they all refer to browser.switchToFrame.
I use @wdio/cli: 7.9.1 and there is no browser.switchToFrame.

So how do I interact with a iframe?

Best regards,
Dag

**Comment by christian-bromann** [comment]
> I use @wdio/cli: 7.9.1 and there is no browser.switchToFrame.

What do you mean by that?

**Reply by daganilsenwork** [reply]
Hi. API->browser.
There is no command called switchToFrame. In previous versions there is.

Many examples on the web like:


// open webpage
browser.url('https://chercher.tech/practice/frames')
// store the text value
var textValue = $("//label/span").getText();
// switch to frame1
browser.switchToFrame("frame1")
// set the value of the textbar to the value stored
$("//input[@type='text']").setValue(textValue)


This is not possible in version 7.x.x
On the API->WebDriver Protocol there is a switchToFrame command. 
How can this work together (we use other commands from browser.)

I'm new to WebdriverIO ;-)

regards,
Dag

**Reply by christian-bromann** [reply]
`switchToFrame` is a valid WebDriver command, see: https://webdriver.io/docs/api/webdriver/#switchtoframe

**Reply by christian-bromann** [reply]
Here is an example that works:
```js
const { remote } = require('webdriverio')

;(async () => {
    const browser = await remote({
        capabilities: {
            browserName: 'chrome'
        }
    })

    await browser.url('https://chercher.tech/practice/frames')
    await browser.switchToFrame(0)
    await browser.switchToFrame(0)
    console.log(await browser.$('html').getText()) // Inner Frame Check box

    await browser.deleteSession()
})().catch(console.error)
```

**Reply by daganilsenwork** [reply]
Hi. Thanks for the example. But how does this work together with the already initiated process.
I already use the global browser object in the running test:

```
describe("Our Login startpage", () => {
  it("should login with valid credentials", async () => {    
    browser.url("ourlogonpage");
    ...
    ...
  });
});
```

regards,
Dag

**Reply by christian-bromann** [reply]
it works the same
